File: client/src/app/modules/miscellaneous/services/customer.order.frequency.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { CustomerOrderFrequency } from '../../miscellaneous/model/CustomerOrderFrequency';

@Injectable({
  providedIn: 'root'
})
export class CustomerOrderFrequencyService extends AppRestService<CustomerOrderFrequency>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getCustomerOrderFrequencies() {
    return this.getListCached(new HttpParams(), "customer-order-frequencies");
  }

  addOrUpdateCustomerOrderFrequency(customerOrderFrequency: CustomerOrderFrequency) {
    return this.addOrUpdate(new HttpParams(), "customer-order-frequency", customerOrderFrequency, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/CustomerOrderFrequencyService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.CustomerOrderFrequency;

public interface CustomerOrderFrequencyService {
    public List<CustomerOrderFrequency> getCustomerOrderFrequencies();

    public CustomerOrderFrequency getCustomerOrderFrequency(Integer id);

    public CustomerOrderFrequency addOrUpdateCustomerOrderFrequency(CustomerOrderFrequency customerOrderFrequency);
}
